src/transformers/utils/quantization_config.py

```diff
@@ -948,7 +948,7 @@ class AqlmConfig(QuantizationConfigMixin):
             Number of codebooks for the Additive Quantization procedure.
         nbits_per_codebook (`int`, *optional*, defaults to 16):
             Number of bits encoding a single codebook vector. Codebooks size is 2**nbits_per_codebook.
-        linear_weights_not_to_quantize (`Optional[List[str]]`, *optional*):
+        modules_to_not_convert (`Optional[List[str]]`, *optional*):
             List of full paths of `nn.Linear` weight parameters that shall not be quantized.
         kwargs (`Dict[str, Any]`, *optional*):
             Additional parameters from which to initialize the configuration object.
```

Comment:
Do you mean I just add a warning that we will move to `modules_to_not_convert` in future versions ?

Reply:
You need to overwrite `from_dict` so that you pop the value of `linear_weights_not_to_quantize` and you set the value of `modules_to_not_convert`. This should do the tricks for serialized models. Then in the init of the config class, you need to deprecate linear_weights_not_to_quantize with a warning that the arg will be deprecated + set `self.modules_to_not_convert = self.linear_weights_not_to_quantize`. You can check gptq config for an example even if it is not the best (e.g `disable_exllama`)